### ISSUE TITLE
Some google update has broken oauth

### ISSUE THREAD

**Issue by elgerg** [mdhom/node-red-contrib-google-oauth-calendar#8]
As usual after a week I have to re-auth to get my calendar items. However this week I get:
Error 400: invalid_request
![image](https://user-images.githubusercontent.com/24628571/156940416-5fae76d1-62be-485e-bd44-8cde10e4403a.png)

Any ideas as to what we can do?

It would be nice to get a permeant grant. Any ideas on that one?

Thanks!

**Comment by elgerg** [comment]
Looks like this:
https://stackoverflow.com/questions/71318804/google-oauth-2-0-failing-with-error-400-invalid-request-for-some-client-id-but
https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#dates-oob

I think a lot more users are going to start getting this :(

**Comment by elgerg** [comment]
@mdhom is this now dead?
Thanks

**Comment by Dirnei** [comment]
No its not dead. We are a bit busy now, but we will have a look the next days

**Comment by elgerg** [comment]
Cool, thanks. Appreciate it.

**Comment by elgerg** [comment]
Hi,
Any updates? Is there anything I can do to help with testing etc?
Thanks!

**Comment by mdhom** [comment]
just published version 1.0.2 with a bumped googleapis version. Can you test if that works @elgerg?
The thing is, the packages doesn't handle the OAuth calls itself but uses the googleapis package for that. By updating, I hope the package complies with the new google rules...

**Comment by elgerg** [comment]
@mdhom I'll give it a go as soon as I can. Any idea how long before the update will appear in the node-red palette or here:
https://flows.nodered.org/node/node-red-contrib-google-oauth-calendar

Thanks!

**Comment by elgerg** [comment]
Sorry, no change. :(
I think it's because google have deprecated urn:ietf:wg:oauth:2.0:oob so it just wont work. It needs a real callback URL now.
![image](https://user-images.githubusercontent.com/24628571/160293373-d54018b6-13ec-4c69-b195-f9b54897e988.png)

**Comment by elgerg** [comment]
Hi,
Any thoughts?
Thanks

**Comment by elgerg** [comment]
Hey,
Anything else I can test?
Thanks!

**Comment by elgerg** [comment]
Hi,

Any updates?

Thanks

**Comment by gpuccio87** [comment]
Hi, I have the same problem. Do you have any updates?

**Comment by elgerg** [comment]
Hi,

Any updates/news? I'd really like to get calendar integration working in my project again.

Thanks

**Comment by Toby-Mills** [comment]
I received a notification email from Google today saying that my application will be blocked on Oct 3rd because it uses the OOB workflow.
Does anybody have any idea how we might resolve this. It seems like you need to have a proper callback URL now.

**Comment by elgerg** [comment]
Sorry to chase but it's been 2 months now with no calendar updates. Should we be looking for an alternative? I'd rather not as I liked the format and usability of this node but if you cant get it working I'm not sure what to do..

Thanks

**Comment by elgerg** [comment]
@mdhom Any news?

**Comment by elgerg** [comment]
Nothing?

**Comment by gpuccio87** [comment]
Is the project abandoned?

**Comment by elgerg** [comment]
On the 15th of March they said it's not abandoned but I dont think we are getting anywhere :(

**Comment by Toby-Mills** [comment]
@mdhom I don't really have any experience with OAuth, but I'm keen to help keep this project alive (it works brilliantly for my needs).
Is there anything I could do to help?

**Comment by elgerg** [comment]
@Dirnei / @mdhom  are you planning on working on this or should we all start looking for another solution?
Thanks

**Comment by elgerg** [comment]
It might be worthwhile looking into these:

https://farmer-eds-shed.com/interacting-with-google-calendar-using-node-red/
https://github.com/Farmer-Eds-Shed/Node-Red-Google-API-Flows

There are some examples on how to use a callback URL.

**Comment by Toby-Mills** [comment]
@elgerg I gave that a try. It looked promising, but when I finally got to the OAuth 'Sign In with Google' dialog, it responded with Error 403: access_denied
Please let me know if you have success with it

**Comment by elgerg** [comment]
@Toby-Mills 
Yes, it worked for me. You do have to remember to add the clientID, secret and the 2 pages into the Oauth2 Client in the google API settings.

**Comment by elgerg** [comment]
Oh, one thing to bear in mind, you need to use 127.0.0.1 (and use the UI from the local machine) or it will give the same 403.

**Comment by elgerg** [comment]
So, it took a little messing around but using the following:
https://github.com/Farmer-Eds-Shed/Node-Red-Google-API-Flows
I have it working as it used to:

![image](https://user-images.githubusercontent.com/24628571/174869124-2ae1a4f5-378a-4c54-8dc1-a3aeafd7e54d.png)

Its obviously a bit more complicated than it used to be but it does work!!

I guess that this project is indeed dead given the lack of developer response.

**Comment by mdhom** [comment]
thanks for your efforts @elgerg. What you built can be a good blueprint for what to implement in this library, so that it works again. Can you share the flow json, so that everyone can reuse your flow and have a deeper look in how that works?

As Christian and I are totally overloaded with work from our company, we can't provide an update anytime soon. But as usual in open source projects, everyone is very welcome to contribute actively.

**Comment by Toby-Mills** [comment]
I'm glad you got it working! I'm still struggling though :(
I am using it from the local machine, using http://127.0.0.1:1880
What did you mean when you said _"You do have to remember to add the clientID, secret and the 2 pages into the Oauth2 Client in the google API settings"_? I did create the application and created a set of Oauth credentials, then used the clientID and  Secret from there in the node-red subflow properties. But still just get the 403

**Comment by elgerg** [comment]
@mdhom, I really just used the examples from Farmer Ed. The only addition was to combine the multiple streams into a single one (that was one of the major advantages of your node).

**Comment by elgerg** [comment]
@Toby-Mills 

Have a look at the below:
![image](https://user-images.githubusercontent.com/24628571/174872224-15316802-eea9-44d0-a6ef-2b0feb2ab185.png)

You need the clientID and secret into the node:
![image](https://user-images.githubusercontent.com/24628571/174872338-0f8dbb7a-3cd0-4ee3-bbd7-d4da280732ee.png)

and the redirect and dashboard.

Give that a go and see how you go.

If you are still struggling it might be worth opening an issue here:
https://github.com/Farmer-Eds-Shed/Node-Red-Google-API-Flows/issues

As this is really related to the Farmer-Eds stuff rather than mdhom..

Ta!

**Comment by Toby-Mills** [comment]
@elgerg I appear to have got it working. When I created a new project in Google Cloud it didn't work (I think because the project is 'unverified'). But then I tried just adding new credentials to my older project, and that seems to have worked!

**Comment by elgerg** [comment]
@Toby-Mills Nice!
